src/main/java/no/sysco/middleware/prometheus/kafka/common/KafkaClientJmxCollector.java

```diff
@@ -0,0 +1,169 @@
+     * */
+    private ObjectName getObjectNameFromString(final String metricType, final String id) {
+        String objectNameWithDomain = domainName + ":type=" + metricType + ",client-id="+id;
+        System.out.println(objectNameWithDomain);
+        ObjectName responseObjectName = null;
+        try {
+            ObjectName mbeanObjectName = new ObjectName(objectNameWithDomain);
+            Set<ObjectName> objectNames = mBeanServer.queryNames(mbeanObjectName, null);
+            for (ObjectName object: objectNames) {
+                responseObjectName = object;
+            }
+        } catch (MalformedObjectNameException mfe) {
+            throw new IllegalArgumentException(mfe.getMessage());
+        }
+        return responseObjectName;
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T extends Number> T getMBeanAttributeValue(final String metricType, final String attribute, final String id, final Class<T> returnType) {
+
+        System.out.println(String.format("domainName:%s ; metricType:%s ; attribute:%s ; client-id:%s", domainName, metricType, attribute, id));
+        ObjectName objectName = getObjectNameFromString(metricType, id);
+        if (objectName == null) {
+            // This also indicates that the mbeanObject is not registered.
+            // Check to see if it is an exceptions-$class object
+            if (metricType.startsWith("exceptions")) {
+                // The exceptions object is not added to the MBeanServer, as there
+                // were no HTTP client exceptions raised yet. Return 0.
+                if (returnType.equals(Double.class)) {
+                    return (T) Double.valueOf(0d);
+                } else if (returnType.equals(Long.class)) {
+                    return (T) Long.valueOf(0L);
+                }
+            }
+            String message = "Requested MBean Object not found";
+            throw new IllegalArgumentException(message);
+        }
+
+        Object value;
+        try {
+            System.out.println("object name: "+ objectName + ", attribute: " + attribute);
+            value = mBeanServer.getAttribute(objectName, attribute);
+            final Number number;
+
+            if (value instanceof Number) {
+                number = (Number) value;
+            } else {
+                try {
+                    number = Double.parseDouble(value.toString());
+                } catch (NumberFormatException e) {
+                    String message = "Failed to parse attribute value to number";
+                    System.out.println(e);
+                    throw new IllegalArgumentException(message);
+                }
+            }
+
+            if (returnType.equals(number.getClass())) {
+                return (T)number;
+            } else if (returnType.equals(Short.class)) {
+                return (T)Short.valueOf(number.shortValue());
```

Comment:
shortValue return already short, is boxing needed?
Same for all other variants

Reply:
yes, it is for type compatibility. Method returns type `<T extends Number> T`, if I would remove explicit casting, it won't compile.